openbb_platform/providers/polygon/openbb_polygon/models/income_statement.py

```diff
@@ -22,41 +23,44 @@ class PolygonIncomeStatementQueryParams(IncomeStatementQueryParams):
 
     __alias_dict__ = {"symbol": "ticker", "period": "timeframe"}
 
-    period: Literal["annual", "quarter", "ttm"] = Field(default="annual")
-    filing_date: Optional[date] = Field(
+    period: Literal["annual", "quarter", "ttm"] = Field(
+        default="annual",
+        json_schema_extra={"choices": ["annual", "quarter", "ttm"]},
```

Comment:
For future reference:
For the CLI to leverage custom choices, one only need to populate the `json_schema_extra` with them if it's not possible to use such options inside the `Literal` - cases where we want to leverage comma separated lists and as such the type need to be `str` instead of `Literal`.
Here, we're just duplicating the same info.

Reply:
From what I've experienced, the act of defining almost any Literal is an exercise in duplication, and this appears to be a limitation of Python 3.9, because you cannot dynamically define a Literal.